packages/api/db/migration/20240110204504_create_animal_type_table.js

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright (c) 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+export const up = async function (knex) {
+  await knex.schema.createTable('animal_type', (table) => {
+    table.increments('id').primary();
+    table.uuid('farm_id').references('farm_id').inTable('farm').defaultTo(null);
+    table.string('type').defaultTo(null);
+    table.string('type_key').defaultTo(null);
+    table.boolean('deleted').notNullable().defaultTo(false);
+    table.string('created_by_user_id').references('user_id').inTable('users');
+    table.string('updated_by_user_id').references('user_id').inTable('users');
+    table.dateTime('created_at').notNullable();
+    table.dateTime('updated_at').notNullable();
```

Comment:
We could add a check constraint to this table to ensure that either `type` or `type_key` are filled in. I think it would be something like this:

`table.check('?? is not null or ?? is not null', ['type', 'type_key']);`

Knex docs for check constraints https://knexjs.org/guide/schema-builder.html#checks

Reply:
Okay awesome thank you, I will add that!